src/vmm_unix.ml

```diff
@@ -374,23 +377,26 @@ let mb_of_bytes size =
 
 let find_block_devices () =
   let blockdir = Fpath.(!dbdir / block_sub) in
-  Bos.OS.Dir.contents ~rel:true blockdir >>= fun files ->
-  List.fold_left (fun acc file ->
-      acc >>= fun acc ->
-      let path = Fpath.append blockdir file in
-      Bos.OS.File.exists path >>= function
-      | false ->
-        Logs.warn (fun m -> m "file %a doesn't exist, but was listed" Fpath.pp path) ;
-        Ok acc
-      | true ->
-        Bos.OS.Path.stat path >>= fun stats ->
-        match mb_of_bytes stats.Unix.st_size, Name.of_string (Fpath.to_string file) with
-        | Error (`Msg msg), _ ->
-          Logs.warn (fun m -> m "file %a size error: %s" Fpath.pp path msg) ;
-          Ok acc
-        | _, Error (`Msg msg) ->
-          Logs.warn (fun m -> m "file %a name error: %s" Fpath.pp path msg) ;
+  Bos.OS.Path.exists blockdir >>= function
+  | false -> Bos.OS.Dir.create blockdir >>| fun _ -> []
+  | true ->
```

Comment:
Maybe replace with:

``` OCaml
(Bos.OS.Path.exists blockdir >>= function
  | false -> Bos.OS.Dir.create blockdir >>| fun _ -> ()
  | true -> Ok ()) >>= fun () ->
```

It adds an unnecessary system call, but the diff is smaller and we catch the (probably highly unlikely) case where someone creates a file inbetween the exists call and create. I don't have strong feelings about this...

Reply:
good point. I guess for safety the case where someone removes the block subdirectory while albatrossd is executing also needs to be handled (when `open` returns ENOENT)

Reply:
now, the create_block creates the block directory if it does not exist -- and uses permissions 0700